### ISSUE TITLE
About insert picture, picture save problem

### ISSUE THREAD

**Issue by rebron1900** [zadam/trilium#2537]
### Describe feature

不知道可不可以设置一些图床用来保存图片？就像sm.ms这种。
如果进行了相关的设置就把图片传到在线服务上面去。

有一些组件库应该可以减少开发的工作，比如：[PicGo](https://github.com/PicGo/PicGo-Core)

thanks，i love trilium

---
I wonder if I can set up some map beds for saving pictures? Like sm.ms.
Upload the image to the online service if you have set it up.

There are some component libraries that should reduce development effort, such as:[PicGo](https://github.com/PicGo/PicGo-Core)

**Comment by zadam** [comment]
Hi, what would be the benefit of this?

I actually consider it to be a great advantage that the pictures is saved locally and available offline.

**Comment by rebron1900** [comment]
从长远来看服务器的储存费用比较贵，因为图片的资源占用比较高，而这些第三方的储存服务可以比较便宜或者免费
In the long run, the storage cost of the server is more expensive because of the high resource consumption of the image, and these third-party storage services can be cheaper or free

**Comment by weihao306** [comment]
I second that. I used markdown as a note method and the map bed as a picture storage. Now I migrate to trilium and have to download pictures from the map bed again and copy them to trilium. In addition, some pictures may be large, the storage cost of the server is more expensive

**Comment by lushaojun** [comment]
我当然赞成图片掌握在自己手中。
但我们有很多其他的方式保存图片。
所以，我也建议提供一个外部链接图片的方式，这样有益于缩减数据库体积。
这样也是等于多一个选择，对使用者来说，也有更大的灵活性。

I certainly agree that the picture should be in my own hands.
But we have many other ways to save pictures.
Therefore, I also suggest providing a way to link pictures externally, which is beneficial to reducing the size of the database.
This is also equal to one more choice and greater flexibility for users.

**Comment by chunjinfeng** [comment]
I wonder whether the images are saved in sqlite? In the long run, as the number of notes bigger and bigger, the pressure on sqlite could bear ?

**Comment by zadam** [comment]
The images are saved in SQLite, yes. It's not really an issue from a scalability point of view, SQLite is often used to store a lot of GBs of data ...

**Comment by lushaojun** [comment]
> The images are saved in SQLite, yes. It's not really an issue from a scalability point of view, SQLite is often used to store a lot of GBs of data ...

这并不是扩展性的问题，而是成本与效益的问题。
很多专业存储公司，存储价格极低，速度极快，能极大减轻图片存储的烦恼。

This is not a matter of scalability, but a matter of cost and benefit.

Many professional storage companies have extremely low storage price and fast speed, which can greatly reduce the trouble of image storage.

**Comment by rebron1900** [comment]
> The images are saved in SQLite, yes. It's not really an issue from a scalability point of view, SQLite is often used to store a lot of GBs of data ...

昨天在一个trilium的交流频道中有人说他的数据库文件已经超过2gb，他很担心文件损坏丢失数据，
我相信里面肯定有很多图片之类的大体积文件。

Yesterday someone on the trilium communications channel said that his database files had exceeded 2GB and he was worried about file corruption and data loss
I'm sure there are lots of pictures and big files in there.

**Comment by chunjinfeng** [comment]
Maybe a lot of images is not the original use of trilium?
If you only take notes of codes and words not pictures,it really good.
Use trilium for take notes ,and not webclipper. Just wirte down your thoughts,your ideas and it will be happy.

**Comment by zadam** [comment]
> Maybe a lot of images is not the original use of trilium?

That's correct. Big file/image storage is not the goal. Trilium also offers compression settings which can lower the image sizes considerably.

But also 2 GB database should be fine. SQLite should scale well up to hundreds of gigabytes.

**Comment by chunjinfeng** [comment]
My friend ,rebron1900 ,maybe want to say that if inserting images could support markdown link of using image links ,It can save a lot disk spaces and will not let the database grow up to 2GBs, maybe 500MB,at the same time,the sync presuress will also be cool down.
What a good thing it is ,isn't it?

**Comment by sigaloid** [comment]
> My friend ,rebron1900 ,maybe want to say that if inserting images could support markdown link of using image links ,It can save a lot disk spaces and will not let the database grow up to 2GBs, maybe 500MB,at the same time,the sync presuress will also be cool down.

It does, at least when importing, save the images (not link to them and load them). I saved the README.md of this repo, imported it, and the images did indeed get downloaded. 

Alternatively, use the "import from markdown" like so: 
![Screenshot from 2022-01-24 21-37-01](https://user-images.githubusercontent.com/69441971/150900611-0870808b-0c47-4fbe-956c-74207ea93d0f.png)

This is not what you're looking for though since it still saves it. Which is a good thing, I would be upset if I lost connection and couldn't see my images. In my opinion, if you're storing gigabytes of images, it's not really the intended use case and there are better applications to use.

**Comment by eliandoran** [comment]
We already have support not to download images (configurable from settings), and Markdown import can be used to import any image. Combined with the upcoming #6546, this is the best we plan to support for the idea of external image storage.
